File: lab1/lab1.py
nums = [99,0,10,1]
target = 100

def findPair(arr, t):
    d = {}
    for i in arr:
        d[i] = t - i

    output = [-1,-1]
    for i in range(len(arr)):
        if d[arr[i]] in d.keys():
            output[0] = i
            for j in range(i+1, len(arr)):
                if arr[j] == d[arr[i]]:
                    output[1] = j
                    return output
    return output

print(findPair(nums, target))

File: lab1/lab1.java
import java.util.HashMap;

//TIP To <b>Run</b> code, press <shortcut actionId="Run"/> or
// click the <icon src="AllIcons.Actions.Execute"/> icon in the gutter.
public class lab1 {
    private static int[] findPair(int[] nums, int target){
        HashMap<Integer, Integer> d = new HashMap<>();
        for (int i : nums) {
            d.put(i, target - i);
        }

        for (int i : nums){
            if (d.containsKey(d.get(i))){
                return new int[] {i, d.get(i)};
            }
        }

        return null;
    }

    public static void main(String[] args) {
        int[] arr = {99, 0, 1, 10};
        int t = 109;

        int[] o = findPair(arr, t);

        System.out.println(o == null?"no pair":o[0] + " " + o[1]);
    }
}